### PR TITLE
Initial commit for LongNeck() class

### PR DESCRIPTION
**Pull request by benmcroberts**

### CODE REVIEW

File: core/traits/long_neck.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from core.constants import Signal, TraitCategory
+from core.trait_card import TraitCard
+from pydispatch import dispatcher
+
+
+class LongNeck(TraitCard):
+    def __init__(self):
+        super().__init__(type(self).__name__, tuple([TraitCategory.EATING]))
```
Comment (a-toms) [inline core/traits/long_neck.py]: For clarity, I suggest: 

a) assigning the two super() arguments to two variables above the super() call. The variables could be called trait_card_name and trait_category; 
b) passing these variables to super(). 

What do you think?

File: core/traits/long_neck.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        dispatcher.connect(self.handler_start_feeding_phase, Signal.START_FEEDING_PHASE)
+
+    def handler_start_feeding_phase(self):
+        # self._owner can be None, otherwise expect it to be a reference.
+        # The reference may resolve to None, so we check for that too.
```
Comment (a-toms) [inline core/traits/long_neck.py:15]: Am I right in thinking that self._owner would be None where the trait card is not yet played as a trait card on a species?

File: core/trait_card.py [hunk 3]
```diff
@@ -18,3 +21,9 @@ def __eq__(self, other: object) -> bool:
     def __hash__(self) -> int:
         return hash(tuple(vars(self).values()))
 
+    def set_owner(self, owner: object):
+        self._owner = weakref.ref(owner)
+
```
Comment (a-toms) [inline core/trait_card.py:27]: Suggest adding a comment to explain what owner is or renaming owner. It took me a while to understand why we are calling self._owner in long_neck.py.

**Review by a-toms — CHANGES_REQUESTED** [review]
I am looking forward to building more trait cards! 😃

**Comment by a-toms** [comment]
I added non-circular type hints to this branch.

File: core/traits/long_neck.py (hunk 1 above)
Comment (benmcroberts) [reply core/traits/long_neck.py]: Makes sense, done.

File: core/traits/long_neck.py (hunk 2 above)
Comment (benmcroberts) [reply core/traits/long_neck.py:15]: Yeah, that's right -- means that we can do tests etc on a TraitCard() even if it's not owned by a species.

File: core/trait_card.py (hunk 3 above)
Comment (benmcroberts) [reply core/trait_card.py:27]: I'll rename it to parent_species

File: core/traits/long_neck.py (hunk 2 above)
Comment (a-toms) [reply core/traits/long_neck.py:15]: Roger!

**Review by a-toms — APPROVED** [review]

**Comment by a-toms** [comment]
Merged and deleted branch.

**Pull request merged**